library/ssl_ticket.c

```diff
@@ -245,8 +245,8 @@ int mbedtls_ssl_ticket_write( void *p_ticket,
     {
          goto cleanup;
     }
-    state_len_bytes[0] = ( clear_len >> 8 ) & 0xff;
-    state_len_bytes[1] = ( clear_len      ) & 0xff;
+    state_len_bytes[0] = MBEDTLS_BYTE_1( clear_len );
+    state_len_bytes[1] = MBEDTLS_BYTE_0( clear_len );
 
```

Comment:
We still have explicit instances of the byte shifting and masking pattern in the code, like in `ssl_tls.c`, `ecjpake.c`, `ssl_srv.c`, `asn1write.c`, `test_suite_asn1parse.function` and `ssl_server2.c`. And I think that adding the `MBEDTLS_BYTE_x` macro up to 7th byte would be useful as well - we have 4 instances of such operations.

Reply:
the remaining instances use an unsigned char. does it make sense to cast it to a `uint8_t` in the macro and then recast it to an `unsigned char` in the general code, or should I define separate macros for `unsigned char`.